src/app/features/automated-status-update/automated-status-update-config-service.ts

```diff
@@ -25,283 +25,370 @@ export class AutomatedStatusUpdateConfigService {
   private dialog = inject(MatDialog);
   private unsavedChangesService = inject(UnsavedChangesService);
   private entitySchemaService = inject(EntitySchemaService);
-  private configService = inject(ConfigService);
 
   /**
    * Track processed entity revisions to prevent duplicate automated status updates within the same save operation
    */
   private readonly processedRevisions = new Set<string>();
 
-  /**
-   * List of entities impacted by automated status updates.
-   * For example, if a field in the "Schools" entity is linked to a field in the "Child" entity,
-   * and the field in "Schools" (trigger field) is updated, the "Child" entity will be affected.
-   * The IDs of the affected "Child" entities will be stored in this list.
-   */
-  relatedEntities: Entity[] = [];
-
   /**
    * The relatedReferenceField configuration, which is used to show the label of related reference field in dialog.
    */
   relatedReferenceFieldConfig: EntitySchemaField;
 
   /**
-   * A map that tracks dependencies between entities and their fields for automated status updates.
-   * - Key: A unique identifier in the format "entityType|fieldId" (e.g., "School|schoolStatus").
-   * - Value: An array of rules that define how changes in the specified field affect related entities and fields.
-   *
-   * This map is used to determine which entities and fields need to be updated automatically
-   * when a specific field in a source entity changes.
+   * Applies rules to dependent entities based on changes in the provided entity.
+   * Also prompts user confirmation and saves updates if any changes were made.
+   * @param entity - The source entity whose changes should trigger updates
+   * @param entityBeforeChanges The state of the entity before changes were applied to identify what has changed
    */
-  private dependencyMap = new Map<
-    string,
-    {
+  public async applyRulesToDependentEntities(
+    entity: Entity,
+    entityBeforeChanges: Entity,
+  ): Promise<void> {
+    if (this.checkRefAlreadyProcessed(entity)) return;
+
+    // (1) get array of changed fields (we later need to compare if these are used as a sourceValueField)
+    const changedFields = this.getChangedFields(entity, entityBeforeChanges);
+    const changedFieldIds = changedFields.map((f) => f.fieldId);
+
+    // (2.1) check schema fields of the entity if any automation rule uses the field as a sourceValueField
+    // [NEW] (2.2) check schema fields of all other entity types, if any automation rule uses this sourceEntityType (= entity.getType()) and sourceValueField
+    const relevantDirectRules: DefaultValueConfigInheritedField[] =
+      this.getInheritanceRulesFromDirectEntity(entity.getConstructor()).filter(
+        (r) => changedFieldIds.includes(r.sourceValueField),
+      );
+    const relevantIndirectRules: Array<{
+      rule: DefaultValueConfigInheritedField;
       targetEntityType: EntityConstructor;
       targetFieldId: string;
```

Comment:
maybe include the whole EntitySchemaField of the target field, if the rule is not enough?